BEST_PRACTICES.md

```diff
@@ -203,3 +216,459 @@ Use a simple NSURLSession wrapper to make things simpler, [Networking](https://g
+}
+```
+
+**Not Preferred:**
+```swift
+var optionalSubview: UIView?
+
+if let unwrappedSubview = optionalSubview {
+    // do something with unwrappedSubview
+}
+```
+
+### Struct Initializers
+
+Use the native Swift struct initializers rather than the legacy CGGeometry constructors.
+
+**Preferred:**
+```swift
+let bounds = CGRect(x: 40.0, y: 20.0, width: 120.0, height: 80.0)
+var centerPoint = CGPoint(x: 96.0, y: 42.0)
+```
+
+**Not Preferred:**
+```swift
+let bounds = CGRectMake(40.0, 20.0, 120.0, 80.0)
+var centerPoint = CGPointMake(96.0, 42.0)
+```
+
+Prefer the struct-scope constants `CGRect.infiniteRect`, `CGRect.nullRect`, etc. over global constants `CGRectInfinite`, `CGRectNull`, etc. For existing variables, you can use the shorter `.zeroRect`.
+
+### Type Inference
+
+The Swift compiler is able to infer the type of variables and constants. You can provide an explicit type via a type alias (which is indicated by the type after the colon), but in the majority of cases this is not necessary.
+
+Prefer compact code and let the compiler infer the type for a constant or variable.
+
+**Preferred:**
+```swift
+let message = "Click the button"
+var currentBounds = computeViewBounds()
+```
+
+**Not Preferred:**
+```swift
+let message: String = "Click the button"
+var currentBounds: CGRect = computeViewBounds()
+```
+
+**NOTE**: Following this guideline means picking descriptive names is even more important than before.
+
+### Syntactic Sugar
+
+Prefer the shortcut versions of type declarations over the full generics syntax.
+
+**Preferred:**
+```swift
+var deviceModels: [String]
+var employees: [Int: String]
+var faxNumber: Int?
+```
+
+**Not Preferred:**
+```swift
+var deviceModels: Array<String>
+var employees: Dictionary<Int, String>
+var faxNumber: Optional<Int>
+```
+
+## Semicolons
+
+Swift does not require a semicolon after each statement in your code. They are only required if you wish to combine multiple statements on a single line.
+
+Do not write multiple statements on a single line separated with semicolons.
+
+## Resources
+
+In `Swift` it's a good practice to use extensions for accessing elements of asset catalogs, custom colors and fonts. It helps to avoid the error-prone practice of hardcoding strings into your code.
+
+Use a `Theme.swift` that contains extensions for theming your app. When adding color extensions avoid adding the `Color` postfix to your color methods, also add some context of how this color is used, just be careful not to have one method per UI element.
+
+Colors should be semantic and context aware, adding the controller or the flow to the color name helps a lot.
+
+**Preferred:**
+```swift
+self.color = .onboardingButtonNormal
+self.highlightedColor = .onboardingButtonHighlighted
+```
+
+**Not Preferred:**
+```swift
+self.color = .buttonNormal
+self.highlightedColor = .buttonHighlighted
+```
+
+For example:
+
+```swift
+import Hex
+import UIKit
+
+extension UIColor {
```

Comment:
No, since this would change the method signature and I wanted to keep it as close as UIKit does.